jakarta.jdt/org.eclipse.lsp4jakarta.jdt.core/src/main/java/org/eclipse/lsp4jakarta/jdt/core/jax_rs/RootResourceClassDiagnosticsCollector.java

```diff
@@ -70,16 +70,31 @@ public void collectDiagnostics(ICompilationUnit unit, List<Diagnostic> diagnosti
                         int maxParams = 0;
                         String className = type.getElementName();
                         Map<IMethod, Integer> constructorParamsMap = new HashMap<IMethod, Integer>();
-                        
+                        boolean hasPublicConstructor = false;
+                        boolean hasPrivateOrProtectedConstructor = false;
                         for (IMethod method : type.getMethods()) {
                             // if a method of a class has the same name as the class, it is a constructor
                             if ((method.getElementName().equals(className)) && (Flags.isPublic(method.getFlags()))) {
                                 int numParams = method.getNumberOfParameters();
+                                hasPublicConstructor = true;
                                 
                                 if (numParams > maxParams) {
                                     maxParams = numParams;
                                 }
                                 constructorParamsMap.put(method, numParams);
+                            }  
+                            if ((method.getElementName().equals(className)) && (Flags.isPrivate(method.getFlags()) || Flags.isProtected(method.getFlags()))) {
+                            	hasPrivateOrProtectedConstructor = true;
```

Comment:
Double check the formatting, looks slightly off here

Configure your Eclipse to use spaces instead of tabs:
https://newbedev.com/how-do-i-change-eclipse-to-use-spaces-instead-of-tabs

Then run the formatter against this class:
https://www.eclipse.org/pdt/help/html/formatting_code.htm#:~:text=Open%20the%20required%20file.,press%20Ctrl%2BShift%2BF